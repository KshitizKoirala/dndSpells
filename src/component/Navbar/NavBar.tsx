import { Container, AppBar } from "@mui/material";

// Show Navbar in all Pages
function NavBar() {
  return (
    <Container maxWidth="xl" className="headerNavbar">
      <AppBar color="primary" className="NavbarPanel">
        <h2>Dungeons and Dragons Spells</h2>
      </AppBar>
    </Container>
  );
}

export default NavBar;
