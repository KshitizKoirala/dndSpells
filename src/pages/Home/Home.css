.NavbarPanel {
  height: 4rem;
  padding-top: 0.5rem;
}

.spells-parent {
  margin-top: 5rem;
  width: 100%;
}

.spells-list {
  width: 100%;
  list-style-type: none;
  display: flex;
  justify-content: center;
  flex-flow: row wrap;
  gap: 8px;
}

.spells-single {
  flex: 0 1 calc(20% - 8px);
  box-shadow: 0 3px 10px rgb(0 0 0 / 0.2);
  border-radius: 20px;
  display: flex;
  justify-content: space-between;
  flex-direction: row;
  min-width: 200px;
}

.spells-single p {
  font-weight: bold;
  text-align: left;
  padding: 1rem 0 0 2rem;
}

.spells-single .view-btn {
  background: #fff;
  outline: none;
  border: none;
  margin-right: -5px;
}

.spells-single .favourite-btn {
  width: 50%;
}

.spells-single:hover {
  background-color: lightgray;
}
