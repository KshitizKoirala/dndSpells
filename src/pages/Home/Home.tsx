import { Container } from "@mui/material";
import SpellsList from "../../component/spells/SpellsList";
import "./Home.css";

function Home() {
  return (
    <Container maxWidth="xl" className="headerNavbar">
      <SpellsList />
    </Container>
  );
}

export default Home;
