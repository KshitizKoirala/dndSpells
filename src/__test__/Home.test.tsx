import { createRoot } from "react-dom/client";
import { cleanup, render, screen } from "@testing-library/react";
import { act } from "react-dom/test-utils";
// import renderer from "react-test-renderer";

import Home from "../pages/Home/Home";
import NavBar from "../component/Navbar/NavBar";

afterEach(cleanup);

describe("home", () => {
  function prepare() {
    const { container } = render(
      // <ProviderMock favorites={[]}>
      <Home />,
      // </ProviderMock>
    );
    return { container };
  }

  it("testing match snapshot", () => {
    const { container } = prepare();
    expect(container).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    const div = document.createElement("div");
    const root = createRoot(div);
    // eslint-disable-next-line testing-library/no-unnecessary-act
    act(() => {
      root.render(<Home />);
    });
  });

  it("renders header title in the document", () => {
    render(<NavBar />);
    const linkElement = screen.getByText(/Dungeons and Dragons Spells/);
    expect(linkElement).toBeInTheDocument();
  });

  it("renders the Spells component", () => {
    render(<Home />);
    const linkElement = screen.getByTestId("spells-list");
    expect(linkElement).toBeInTheDocument();
  });
});
