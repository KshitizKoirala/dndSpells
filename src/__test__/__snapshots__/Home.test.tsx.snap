// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home testing match snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXl headerNavbar css-19r6kue-MuiContainer-root"
  >
    <div
      class="spells-parent"
      data-testid="spells-list"
    >
      <ul
        class="mt-3 spells-list"
      />
    </div>
  </div>
</div>
`;
